internal/fs/vss_windows.go

```diff
@@ -0,0 +1,986 @@
+	VSS_E_VOLUME_NOT_SUPPORTED:                "VSS_E_VOLUME_NOT_SUPPORTED",
+	VSS_E_VOLUME_NOT_SUPPORTED_BY_PROVIDER:    "VSS_E_VOLUME_NOT_SUPPORTED_BY_PROVIDER",
+	VSS_E_UNEXPECTED:                          "VSS_E_UNEXPECTED",
+	VSS_E_UNEXPECTED_PROVIDER_ERROR:           "VSS_E_UNEXPECTED_PROVIDER_ERROR",
+	VSS_E_UNSELECTED_VOLUME:                   "VSS_E_UNSELECTED_VOLUME",
+	VSS_E_CANNOT_REVERT_DISKID:                "VSS_E_CANNOT_REVERT_DISKID",
+	VSS_E_INVALID_XML_DOCUMENT:                "VSS_E_INVALID_XML_DOCUMENT",
+	VSS_E_OBJECT_ALREADY_EXISTS:               "VSS_E_OBJECT_ALREADY_EXISTS",
+	FSRVP_E_SHADOW_COPY_SET_IN_PROGRESS:       "FSRVP_E_SHADOW_COPY_SET_IN_PROGRESS",
+}
+
+// Str converts a HRESULT to a human readable string.
+func (h HRESULT) Str() string {
+	if i, ok := hresultToString[h]; ok {
+		return i
+	} else {
+		return "UNKNOWN"
+	}
+}
+
+// VssError encapsulates errors retruned from calling VSS api.
+type vssError struct {
+	text    string
+	hresult HRESULT
+}
+
+// NewVssError creates a new VSS api error.
+func newVssError(text string, hresult HRESULT) error {
+	return &vssError{text: text, hresult: hresult}
+}
+
+// NewVssError creates a new VSS api error.
+func newVssErrorIfResultNotOK(text string, hresult HRESULT) error {
+	if hresult != S_OK {
+		return newVssError(text, hresult)
+	}
+	return nil
+}
+
+// Error implements the error interface.
+func (e *vssError) Error() string {
+	return fmt.Sprintf("VSS error: %s:   %s (%#x)", e.text, e.hresult.Str(), e.hresult)
+}
+
+// VssError encapsulates errors retruned from calling VSS api.
+type vssTextError struct {
+	text string
+}
+
+// NewVssTextError creates a new VSS api error.
+func newVssTextError(text string) error {
+	return &vssTextError{text: text}
+}
+
+// Error implements the error interface.
+func (e *vssTextError) Error() string {
+	return fmt.Sprintf("VSS error: %s", e.text)
+}
+
+// VssContext is a custom type for the windows api VssContext type.
+type VssContext uint
+
+// VssContext constant values necessary for using VSS api.
+const (
+	VSS_CTX_BACKUP VssContext = iota
+	VSS_CTX_FILE_SHARE_BACKUP
+	VSS_CTX_NAS_ROLLBACK
+	VSS_CTX_APP_ROLLBACK
+	VSS_CTX_CLIENT_ACCESSIBLE
+	VSS_CTX_CLIENT_ACCESSIBLE_WRITERS
+	VSS_CTX_ALL
+)
+
+// VssBackup is a custom type for the windows api VssBackup type.
+type VssBackup uint
+
+// VssBackup constant values necessary for using VSS api.
+const (
+	VSS_BT_UNDEFINED VssBackup = iota
+	VSS_BT_FULL
+	VSS_BT_INCREMENTAL
+	VSS_BT_DIFFERENTIAL
+	VSS_BT_LOG
+	VSS_BT_COPY
+	VSS_BT_OTHER
+)
+
+// VssObjectType is a custom type for the windows api VssObjectType type.
+type VssObjectType uint
+
+// VssObjectType constant values necessary for using VSS api.
+const (
+	VSS_OBJECT_UNKNOWN VssObjectType = iota
+	VSS_OBJECT_NONE
+	VSS_OBJECT_SNAPSHOT_SET
+	VSS_OBJECT_SNAPSHOT
+	VSS_OBJECT_PROVIDER
+	VSS_OBJECT_TYPE_COUNT
+)
+
+// UUID_IVSS defines the GUID of IVssBackupComponents.
+var UUID_IVSS = &ole.GUID{
+	0x665c1d5f, 0xc218, 0x414d,
```

Comment:
`go vet` don't like this. Should be `var UUID_IVSS = ole.NewGUID("{665c1d5f-c218-414d-a05d-7fef5f9d5c86}")`

Reply:
Fixed in https://github.com/restic/restic/pull/2274/commits/97322db98cdc31a9aba2c358e59809418cdf68be